cassandra/src/cassandra/core.clj

```diff
@@ -272,16 +272,18 @@
   (let [ex (:exception (ex-data e))
         exception-class (class ex)]
     (condp = exception-class
-      WriteTimeoutException (if-let [write-type (.getWriteType ex)]
-                              (if (or (= write-type WriteType/CAS)
-                                      (= write-type WriteType/BATCH_LOG))
-                                (assoc op :type :info :value :write-timed-out)
-                                (if (or (= write-type WriteType/SIMPLE)
-                                        (= write-type WriteType/BATCH))
-                                  (assoc op :type :ok)
-                                  (assoc op
-                                         :type :fail
-                                         :error :write-timed-out)))
+      WriteTimeoutException (condp = (.getWriteType ex)
+                              WriteType/CAS (assoc op
+                                                   :type :info
+                                                   :value :write-timed-out)
+                              WriteType/BATCH_LOG (assoc op
+                                                         :type :info
+                                                         :value :write-timed-out)
+                              WriteType/SIMPLE (assoc op :type :ok)
```

Comment:
Probably I don't understand it correctly, but why is it `ok` ?

Reply:
Thank you for the question.
In CAS case, the written value will be read by the following read because the CAS write has failed in the commit phase.
But, in other write cases (writes for `map` and `set` in our Jepsen tests), I don't know the WriteType of WriteTimeoutException.
If the WriteType is SIMPLE () in these cases, I need to fix it.
Let me check...

Reply:
The WriteType is `SIMPLE` in `map` and `set` tests.
I modified the WriteTimeoutException handling to check if the write is conditional or not.
Only if the write is CAS (conditional write), the result will be `:ok` after the WriteTimeoutException. Otherwise, it will be `:info`.